src/teachy_tv.c

```diff
@@ -0,0 +1,1797 @@
+                gTeachyTV_StaticResources.optionChosen = input;
+                DestroyListMenu(data[0], &gTeachyTV_StaticResources.scrollOffset, &gTeachyTV_StaticResources.selectedRow);
+                TeachyTvClearWindowRegs();
+                ClearWindowTilemap(1);
+                schedule_bg_copy_tilemap_to_vram(0);
+                TeachyTvRemoveScrollIndicatorArrowPair();
+                data[3] = 0;
+                data[2] = 0;
+                gTasks[taskId].func = TeachyTvRenderMsgAndSwitchClusterFuncs;
+                break;
+            }
+        }
+    }
+}
+
+void TeachyTvClusFuncTransitionRenderBg2TeachyTvGraphicInitNpcPos(u8 taskId)
+{
+    s16 *data = gTasks[taskId].data;
+    TeachyTvBg2AnimController();
+    if (++data[2] > 63)
+    {
+        CopyToBgTilemapBufferRect_ChangePalette(2, gUnknown_203F450->buffer4, 0, 0, 0x20, 0x20, 0x11);
+        TeachyTvSetSpriteCoordsAndSwitchFrame(((u8 *)data)[2], 8, 0x38, 7);
+        schedule_bg_copy_tilemap_to_vram(2);
+        data[2] = 0;
+        ++data[3];
+        PlayNewMapMusic(BGM_FRLG_FOLLOW_ME);
+    }
+}
+
+void TeachyTvClusFuncClearBg2TeachyTvGraphic(u8 taskId)
+{
+    s16 *data = gTasks[taskId].data;
+    if (++data[2] == 134)
+    {
+        FillBgTilemapBufferRect_Palette0(2, 0, 2, 1, 0x1A, 0xC);
+        schedule_bg_copy_tilemap_to_vram(2);
+        data[2] = 0;
+        ++data[3];
+    }
+}
+
+void TeachyTvClusFuncNpcMoveAndSetupTextPrinter(u8 taskId)
+{
+    s16 * data = gTasks[taskId].data;
+    struct Sprite * spriteAddr = &gSprites[data[1]];
+    if (data[2] != 35)
+        ++data[2];
+    else {
+        if (spriteAddr->pos2.x == 0x78)
+        {
+            StartSpriteAnim(&gSprites[data[1]], 0);
+            TeachyTvInitTextPrinter(gUnknown_841B83D);
+            data[2] = 0;
+            ++data[3];
+        }
+        else
+            ++spriteAddr->pos2.x;
+    }   
+}
+
+void TeachyTvClusFuncIdleIfTextPrinterIsActive(u8 taskId)
+{
+    s16 *data = gTasks[taskId].data;
+    if (!sub_80BF518(0))
+        ++data[3];
+}
+
+void TeachyTvRenderMsgAndSwitchClusterFuncs(u8 taskId)
+{
+    s16 *data = gTasks[taskId].data;
+    if (JOY_NEW(B_BUTTON))
+    {
+        u8 *offset = &(gUnknown_203F450->var_4006);
+        u32 zero = 0;
+        offset[0] = 1;
+        TeachyTvSetSpriteCoordsAndSwitchFrame(*((char *)data + 2), 0, 0, 0);
+        FillWindowPixelBuffer(0, 0xCC);
+        CopyWindowToVram(0, 2);
+        TeachyTvClearBg1EndGraphicText();
+        data[2] = zero;
+        data[3] = zero;
+        gTasks[taskId].func = TeachyTvBackToOptionList;
+    }
+    else
+    {
+        void (* const * const *array)(u8) = gTeachyTvFuncClusterArray;
+        void (* const * const cluster)(u8) = array[(u8)gTeachyTV_StaticResources.optionChosen];
+        cluster[data[3]](taskId);
+    }
+}
+
+void TeachyTvClusFuncTextPrinterSwitchStringByOptionChosen(u8 taskId)
+{
+    s16 *data = gTasks[taskId].data;
+    const u8 * const* texts = gTeachyTvTextArray1;
```

Comment:
The definition of `gTeachyTvTextArray1` at line 431 should be moved to here and assigned the name `texts`, and the subsequent rodata definitions should be moved below this function.